modules/queue/manager.go

```diff
@@ -0,0 +1,216 @@
+	mutex sync.Mutex
+
+	counter int64
+	Queues  map[int64]*Description
+}
+
+// Description represents a working queue inheriting from Gitea.
+type Description struct {
+	mutex           sync.Mutex
+	QID             int64
+	Queue           Queue
+	Type            Type
+	Name            string
+	Configuration   interface{}
+	ExemplarType    string
+	addWorkers      func(number int, timeout time.Duration) context.CancelFunc
+	numberOfWorkers func() int
+	counter         int64
+	PoolWorkers     map[int64]*PoolWorkers
+}
+
+// DescriptionList implements the sort.Interface
+type DescriptionList []*Description
+
+// PoolWorkers represents a working queue inheriting from Gitea.
+type PoolWorkers struct {
+	PID        int64
+	Workers    int
+	Start      time.Time
+	Timeout    time.Time
+	HasTimeout bool
+	Cancel     context.CancelFunc
+}
+
+// PoolWorkersList implements the sort.Interface
+type PoolWorkersList []*PoolWorkers
+
+func init() {
+	_ = GetManager()
+}
+
+// GetManager returns a Manager and initializes one as singleton if there's none yet
+func GetManager() *Manager {
+	if manager == nil {
+		manager = &Manager{
+			Queues: make(map[int64]*Description),
+		}
+	}
+	return manager
+}
+
+// Add adds a queue to this manager
+func (m *Manager) Add(queue Queue,
+	t Type,
+	configuration,
+	exemplar interface{},
+	addWorkers func(number int, timeout time.Duration) context.CancelFunc,
+	numberOfWorkers func() int) int64 {
+
+	cfg, _ := json.Marshal(configuration)
+	desc := &Description{
+		Queue:           queue,
+		Type:            t,
+		Configuration:   string(cfg),
+		ExemplarType:    reflect.TypeOf(exemplar).String(),
+		PoolWorkers:     make(map[int64]*PoolWorkers),
+		addWorkers:      addWorkers,
+		numberOfWorkers: numberOfWorkers,
+	}
+	m.mutex.Lock()
+	m.counter++
+	desc.QID = m.counter
+	desc.Name = fmt.Sprintf("queue-%d", desc.QID)
+	if named, ok := queue.(Named); ok {
+		desc.Name = named.Name()
+	}
+	m.Queues[desc.QID] = desc
+	m.mutex.Unlock()
+	log.Trace("Queue Manager registered: %s (QID: %d)", desc.Name, desc.QID)
+	return desc.QID
+}
+
+// Remove a queue from the Manager
+func (m *Manager) Remove(qid int64) {
+	m.mutex.Lock()
+	delete(m.Queues, qid)
+	m.mutex.Unlock()
+	log.Trace("Queue Manager removed: QID: %d", qid)
+
+}
+
+// GetDescription by qid
+func (m *Manager) GetDescription(qid int64) *Description {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	return m.Queues[qid]
+}
+
+// Descriptions returns the queue descriptions
+func (m *Manager) Descriptions() []*Description {
+	m.mutex.Lock()
+	descs := make([]*Description, 0, len(m.Queues))
+	for _, desc := range m.Queues {
+		descs = append(descs, desc)
+	}
+	m.mutex.Unlock()
+	sort.Sort(DescriptionList(descs))
+	return descs
+}
+
+// Workers returns the poolworkers
+func (q *Description) Workers() []*PoolWorkers {
+	q.mutex.Lock()
+	workers := make([]*PoolWorkers, 0, len(q.PoolWorkers))
+	for _, worker := range q.PoolWorkers {
+		workers = append(workers, worker)
```

Comment:
Doesn't golang have something like...?:
```suggestion
		workers = append(workers, ...q.PoolWorkers)
```

Reply:
`q.PoolWorkers` is a map - they don't have that.